src/openfermion/utils/_bch_expansion.py

```diff
@@ -0,0 +1,157 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""Module to efficiently compute the Baker-Campbell-Hausdorff formula."""
+
+import itertools
+import numpy as np
+from scipy.misc import comb, factorial
+
+from openfermion.utils import commutator
+
+def bch_expand(x, y, order):
+    """Compute log[e^x e^y] using the Baker-Campbell-Hausdorff formula
+    Args:
+        x: An operator for which multiplication and addition are supported.
+            For instance, a QubitOperator, FermionOperator or numpy array.
```

Comment:
Really minor: I think you should replace the phrase "numpy array" with "Scipy sparse matrix", because multiplication for Numpy arrays is actually element-wise. Your code might still work for Numpy arrays though because the `commutator` function will support them.